quantum/encoder.c

```diff
@@ -78,14 +79,16 @@ void encoder_init(void) {
 }
 
 static void encoder_update(int8_t index, uint8_t state) {
-    encoder_value[index] += encoder_LUT[state & 0xF];
-    if (encoder_value[index] >= ENCODER_RESOLUTION) {
-        encoder_update_kb(index, false);
-    }
-    if (encoder_value[index] <= -ENCODER_RESOLUTION) { // direction is arbitrary here, but this clockwise
+    encoder_pulses[index] += encoder_LUT[state & 0xF];
+    if (encoder_pulses[index] >= ENCODER_RESOLUTION) {
+        encoder_value[index]++;
         encoder_update_kb(index, true);
     }
-    encoder_value[index] %= ENCODER_RESOLUTION;
+    if (encoder_pulses[index] <= -ENCODER_RESOLUTION) { // direction is arbitrary here, but this clockwise
+        encoder_value[index]--;
+        encoder_update_kb(index, false);
+    }
+    encoder_pulses[index] %= ENCODER_RESOLUTION;
 }
```

Comment:
Just to confirm I'm not crazy, but with this patch it seems to reverse my encoder behavior on preonic rev3 (i.e. clockwise becomes counterclockwise and vice versa).
I tried to reverse true/false in encoder_update_kb part and it seems to fixed it.
If anyone can confirm this on their board I'm happy to create a simple PR.

Reply:
Same w/ a rev6 planck.

Reply:
That's probably my fault, I thought I broke the behavior while doing this work so I make certain my encoder was turning clockwise, and also triggering clockwise (ie: clockwise param as true in this case). So it sounds like different encoders or wiring can affect if it is actually clockwise or not. I'll open a new PR to fix this later tonight.

Reply:
I'm not sure that this should be fixed. 

I checked, yes, this PR does cause the handling to be reversed ....  But I checked it out... it now handles clockwise and counterclockwise CORRECTLY on the planck rev6 (and I'm guessing on the Preonic Rev3). 

Eg, for my encoder, I have:
```c
            if (clockwise)
                tap_code(KC_DOWN);
            else
                tap_code(KC_UP);
```

Prior to this, rotating it counterclockwise would tap down, and clockwise would tap up.  The opposite of what the function is *supposed* to be doing. 

That said, I do think that this is a behavioral change, but I don't think it's something that should be "fixed" at the keyboard level, but rather, at the keymap level.

Reply:
This is really interesting.
Let me grab my multimeter and measure it today.
Perhaps it is caused by the encoder I'm using and it may output wrong signals.

Having said that, perhaps it is not a bad idea to introduce sort of macro to let user define in their own keymaps to reverse the behaviour so it is more clear on keymap code side.

Reply:
I wanted to add that this change now reversed the encoders on my BDN9 keymap, but the encoders work correctly if I compile and flash the default keymap.

Reply:
I also tried the default keymap for planck, same results, it's reversed. Encoder is the one that was listed at its PCB page, from sparkfun. So I think there should be a keymap specific setting to be able to reverse the direction at least.
The comment at line 87 about that branch is clockwise in encoder.c is off after this change for sure.

Reply:
Well, worst case, you can undef the pad declarations, and redefine them in the opposite order. That would reverse the direction that they're registering.

Reply:
Because I don't feel that reverting the behavior is the correct course of action here (eg, the encoders are now actually processing CW and CCW correctly), but to maintain the older behavior, I've opened: 
#7413 

This should allow the old behavior, by simply swapping the pins for you.